src/provable/curve-bigint.ts

```diff
@@ -1,14 +1,34 @@
+import { Ledger } from '../snarky.js';
+import { base58, compose, withVersionNumber } from './binable.js';
 import { Bool, Field } from './field-bigint.js';
 import { provable } from './provable-bigint.js';
 
 export { PublicKey };
 
 type PublicKey = { x: Field; isOdd: Bool };
 
+// TODO generate
+let FIELD_VERSION = 1;
+let PUBLIC_KEY_VERSION = 1;
+
+let FieldWithVersion = withVersionNumber(Field, FIELD_VERSION);
+let BinablePublicKey = withVersionNumber(
+  compose([FieldWithVersion, Bool]),
+  PUBLIC_KEY_VERSION
+);
+let Base58PublicKey = base58(
```

Comment:
base58 encoding of public keys, which is used to implement toJSON / fromJSON for public keys